### ISSUE TITLE
Bug: [@typescript-eslint/no-duplicate-enum-values] Should not emit a warning - in V8

### ISSUE THREAD

**Issue by AntoLC** [typescript-eslint/typescript-eslint#9823]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

*

### Repro Code

Only since v8.*, v7.13.1 is working fine.

This enum create an error when it shouldn't.

```TypeScript
export enum TemplatesOrdering {
  BY_CREATED_ON = 'created_at',
  BY_CREATED_ON_DESC = '-created_at',
}
```
```
Cannot read properties of undefined (reading 'members')
Occurred while linting /src/features/.../useTemplates.tsx:12
Rule: "@typescript-eslint/no-duplicate-enum-values"
error Command failed with exit code 1.
```

### ESLint Config

```javascript
"@typescript-eslint/eslint-plugin": "8.1.0"
```

### Expected Result

I expect to not emit a error.

### Actual Result

Cannot read properties of undefined (reading 'members')
Occurred while linting /src/features/.../useTemplates.tsx:12
Rule: "@typescript-eslint/no-duplicate-enum-values"

### Additional Info

https://github.com/numerique-gouv/impress/commit/905000550de0cbaa20187aa2d2e11d95c06a2232

**Comment by bradzacher** [comment]
https://github.com/numerique-gouv/impress/blob/d76a63be83427c144fbc158208a3e35834231d43/src/frontend/yarn.lock#L4962-L4986

This is working as expected.
Your bot updated the plugin but not the parser.
It's expected that things crash if the two aren't on the same major.

**Comment by AntoLC** [comment]
Yes you're right. 
Thank you for your time.
